src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -2536,306 +2536,318 @@ mgp_error mgp_graph_get_vertex_by_id(mgp_graph *graph, mgp_vertex_id id, mgp_mem
-                                    return impl->DropExistenceConstraint(label_id, property_id);
-                                  },
-                                  [label_id, property_id](memgraph::query::SubgraphDbAccessor *impl) {
-                                    return impl->GetAccessor()->DropExistenceConstraint(label_id, property_id);
-                                  }},
-      graph->impl);
-  if (exist_res.HasError()) {
-    *result = 0;
-  }
-  return mgp_error::MGP_ERROR_NO_ERROR;
+  return WrapExceptions(
+      [graph, label, property]() {
+        auto label_id = std::visit([label](auto *impl) { return impl->NameToLabel(label); }, graph->impl);
+        auto property_id = std::visit([property](auto *impl) { return impl->NameToProperty(property); }, graph->impl);
+        const auto exist_res = std::visit(
+            memgraph::utils::Overloaded{[label_id, property_id](memgraph::query::DbAccessor *impl) {
+                                          return impl->DropExistenceConstraint(label_id, property_id);
+                                        },
+                                        [label_id, property_id](memgraph::query::SubgraphDbAccessor *impl) {
+                                          return impl->GetAccessor()->DropExistenceConstraint(label_id, property_id);
+                                        }},
+            graph->impl);
+        return exist_res.HasError() ? 0 : 1;
+      },
+      result);
 }
 
 mgp_error mgp_list_all_existence_constraints(mgp_graph *graph, mgp_memory *memory, mgp_list **result) {
-  const auto constraint_res =
-      std::visit(memgraph::utils::Overloaded{
-                     [](memgraph::query::DbAccessor *impl) { return impl->ListAllConstraints().existence; },
-                     [](memgraph::query::SubgraphDbAccessor *impl) {
-                       return impl->GetAccessor()->ListAllConstraints().existence;
-                     }},
-                 graph->impl);
-
-  if (auto err = mgp_list_make_empty(constraint_res.size(), memory, result); err != mgp_error::MGP_ERROR_NO_ERROR) {
-    return err;
-  }
+  return WrapExceptions([graph, memory, result]() {
+    const auto constraint_res =
+        std::visit(memgraph::utils::Overloaded{
+                       [](memgraph::query::DbAccessor *impl) { return impl->ListAllConstraints().existence; },
+                       [](memgraph::query::SubgraphDbAccessor *impl) {
+                         return impl->GetAccessor()->ListAllConstraints().existence;
+                       }},
+                   graph->impl);
 
-  for (const auto &label_property_pair : constraint_res) {
-    if (auto err = create_mgp_list_of_label_property(graph, memory, result, label_property_pair);
-        err != mgp_error::MGP_ERROR_NO_ERROR) {
-      return err;
+    if (auto err = mgp_list_make_empty(constraint_res.size(), memory, result); err != mgp_error::MGP_ERROR_NO_ERROR) {
+      throw std::logic_error("Cannot create list");
```

Comment:
c-p of comment from other place: 'I think we need to be more descriptive here: "Listing all unique constraints failed due to failure to create list", or something like that'